testing/src/test/java/io/stargate/it/MultipleStargateInstancesTest.java

```diff
@@ -0,0 +1,144 @@
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metrics.DefaultNodeMetric;
+import com.datastax.oss.driver.internal.core.loadbalancing.DcInferringLoadBalancingPolicy;
+import io.stargate.it.storage.ClusterConnectionInfo;
+import java.lang.reflect.Method;
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import net.jcip.annotations.NotThreadSafe;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+@NotThreadSafe
+public class MultipleStargateInstancesTest extends BaseOsgiIntegrationTest {
+
+  private String table;
+
+  private String keyspace;
+
+  private CqlSession session;
+
+  private int runningStargateNodes;
+
+  public MultipleStargateInstancesTest(ClusterConnectionInfo backend) {
+    super(backend);
+  }
+
+  @BeforeEach
+  public void setup(TestInfo testInfo) {
+    DriverConfigLoader loader =
+        DriverConfigLoader.programmaticBuilder()
+            .withBoolean(DefaultDriverOption.METADATA_TOKEN_MAP_ENABLED, false)
+            .withString(
+                DefaultDriverOption.LOAD_BALANCING_POLICY_CLASS,
+                DcInferringLoadBalancingPolicy.class.getName())
+            .withDuration(DefaultDriverOption.REQUEST_TIMEOUT, Duration.ofSeconds(5))
+            .withDuration(DefaultDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, Duration.ofSeconds(5))
+            .withDuration(DefaultDriverOption.CONTROL_CONNECTION_TIMEOUT, Duration.ofSeconds(5))
+            .withDuration(DefaultDriverOption.REQUEST_TRACE_INTERVAL, Duration.ofSeconds(1))
+            .withStringList(
+                DefaultDriverOption.METRICS_NODE_ENABLED,
+                Collections.singletonList(DefaultNodeMetric.CQL_MESSAGES.getPath()))
+            .build();
+
+    CqlSessionBuilder cqlSessionBuilder = CqlSession.builder().withConfigLoader(loader);
+    for (String host : getStargateHosts()) {
+      cqlSessionBuilder.addContactPoint(new InetSocketAddress(host, 9043));
+    }
+    session = cqlSessionBuilder.build();
+
+    Optional<String> name = testInfo.getTestMethod().map(Method::getName);
+    assertThat(name).isPresent();
+    String testName = name.get();
+    keyspace = "ks_" + testName;
+    table = testName;
+    runningStargateNodes = stargateStarters.size();
+  }
+
+  @Test
+  public void shouldConnectToMultipleStargateNodes() {
+    List<Row> all = session.execute("SELECT * FROM system.peers").all();
+    // system.peers should have N records (all stargate nodes - 1)
+    assertThat(all.size()).isEqualTo(runningStargateNodes - 1);
+  }
+
+  @Test
+  public void shouldDistributeTrafficUniformly() {
+    // given
+    createKeyspaceAndTable();
+    long totalNumberOfRequests = 300;
+    long numberOfRequestPerNode = totalNumberOfRequests / runningStargateNodes;
```

Comment:
nit: would it be more robust from the math perspective to fix `numberOfRequestPerNode` and compute `totalNumberOfRequests = numberOfRequestPerNode * runningStargateNodes`?

Reply:
I totally agree, unfortunately, there are still sometimes cases when one of the Startate nodes does not start (schema agreement problem, timeout). In such a case this test will pass asserting the traffic amongst two nodes, whereas the fixed approach won't. So I've done a less strict approach to not introduce a non-deterministic test. I don't have more ideas on how to make the Stargate startup more robust, I will welcome any inputs.

Reply:
sorry, I do not understand... the numbers should be the same regardless of how well the nodes run, it's just a matter of rounding errors in case the division is not exact.